services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/device/TableManagementStore.java

```diff
@@ -887,44 +956,67 @@ private List<CommonCredential> parseCredentials(final ResultSet result) {
     /**
      * Gets a list of devices of a specific tenant.
      *
-     * @param tenantId the tenantId to search devices
-     * @param pageSize the page size
-     * @param pageOffset the page offset
+     * @param tenantId The tenantId to search devices.
+     * @param pageSize The page size.
+     * @param pageOffset The page offset.
+     * @param filters The list of filters.
      * @param spanContext The span to contribute to.
-     * @return A future containing devices
+     * @return A future containing devices.
      */
-    public Future<SearchResult<DeviceWithId>> findDevices(final String tenantId, final int pageSize, final int pageOffset,
-            final SpanContext spanContext) {
-
+    public Future<SearchResult<DeviceWithId>> findDevices(final String tenantId, final int pageSize, final int pageOffset, final List<Filter> filters,
```

Comment:
given that we only support a single Filter expression, wouldn't it make more sense to use `Optional<Filter>` as the parameter type to make this more explicit?

Reply:
As we intend to support multiple filter expressions in the future, we would like to stay with the signature as is.